File: meal_saver_phone/meal_saver_phone/lib/models/recipe_video_dto.dart
class RecipeVideoDTO {
  final int id;
  final String videoUrl;
  final List<String> tags;
  final String description;
  final String thumbnailUrl;
  final int likes;
  final String username;
  final bool likedByUser;

  RecipeVideoDTO({
    required this.id,
    required this.videoUrl,
    required this.tags,
    required this.description,
    required this.thumbnailUrl,
    required this.likes,
    required this.username,
    this.likedByUser = false, // <- esențial
  });

  factory RecipeVideoDTO.fromJson(Map<String, dynamic> json) {
    return RecipeVideoDTO(
      id: json['id'],
      videoUrl: json['videoUrl'],
      tags: List<String>.from(json['tags'] ?? []),
      description: json['description'],
      thumbnailUrl: json['thumbnailUrl'],
      likes: json['likes'],
      username: json['username'],
      likedByUser: json['likedByUser'] ?? false,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'videoUrl': videoUrl,
    'tags': tags,
    'description': description,
    'thumbnailUrl': thumbnailUrl,
    'likes': likes,
    'username': username,
  };
}

File: MealSaver/src/main/java/com/marcu/mealsaver/Dto/RecipeVideoDTO.java
package com.marcu.mealsaver.Dto;

import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RecipeVideoDTO {
    private Long id;
    private String videoUrl;
    private List<String> tags;
    private String description;
    private String thumbnailUrl;
    private Date createdAt;
    private int likes;
    private String username;
    private boolean likedByUser;

}